src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicListener.cs

```diff
@@ -223,47 +227,47 @@ private unsafe Task StopAsync()
             if (_state.Handle != null)
             {
                 Debug.Assert(!Monitor.IsEntered(_state), "!Monitor.IsEntered(_state)");
-                MsQuicApi.Api.ApiTable->ListenerStop(_state.Handle.QuicHandle);
+                MsQuicApi.Api.ListenerStopDelegate(_state.Handle);
             }
             return _state.StopCompletion.Task;
         }
 
-#pragma warning disable CS3016
-        [UnmanagedCallersOnly(CallConvs = new Type[] { typeof(CallConvCdecl) })]
```

Comment:
Assuming the assertion was:
> condition `out_obj' not met

according to https://github.com/dotnet/runtime/pull/68851#issuecomment-1118918525, then it is tracked by https://github.com/dotnet/runtime/issues/57361 and https://github.com/dotnet/runtime/issues/52977. Perhaps the fix is to use non-array argument in the attribute:

```diff
- [UnmanagedCallersOnly(CallConvs = new Type[] { typeof(CallConvCdecl) })]
+ [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
```
here and in MsQuicStream.cs.

Reply:
It would also require reverting back to using delegates in number of places, partially defeating the point of this change.

I think we either need to get the Mono bug fixed soon; or disable Mono AOT for the Quic library.

Reply:
Agreed. cc @lambdageek (another example where https://github.com/dotnet/runtime/issues/57361 and https://github.com/dotnet/runtime/issues/52977 are blocking array argument usage)